src/components/tasks/TaskList/TaskList.tsx

```diff
@@ -0,0 +1,63 @@
+import { useState } from 'react'
+import Card from '../card'
+import task from '@/interfaces/task.type'
+import beautifyTaskStatus from '@/helperFunctions/beautifyTaskStatus'
+import {
+  SEE_MORE,
+  INITIAL_TASKS_LIMIT,
+  ADD_MORE_TASKS_LIMIT,
+} from '../constants'
+import styles from '../card/card.module.scss'
+
+type TaksListProps = {
+  tasks: task[];
+  isEditable?: boolean;
+  hasLimit?: boolean;
+  updateCardContent?: (id: string, cardDetails: task) => void;
+};
+
+function getFilteredTasks({hasLimit, tasksLimit, tasks}: {hasLimit: boolean, tasksLimit: number, tasks: task[]}) {
+  if (!hasLimit) return tasks;
+  return tasks.slice(0, tasksLimit);
+}
+
+export default function TaskList({
+  tasks,
+  updateCardContent,
+  isEditable = false,
+  hasLimit = false,
+}: TaksListProps) {
+  const initialTasksLimit = hasLimit ? INITIAL_TASKS_LIMIT : tasks.length;
+  const beautifiedTasks = beautifyTaskStatus(tasks);
+  const [tasksLimit, setTasksLimit] = useState<number>(initialTasksLimit);
+  const filteredTasks = getFilteredTasks({tasks: beautifiedTasks, hasLimit, tasksLimit});
```

Comment:
i dont know the full context but do check wether or not we can add memoization in this

Reply:
Using `memoization` here will be over, because state value will always be changing, 
and `see more` button will be unmounted once all tasks are rendered.